feat(creds): support aws-sso configuration

joeybaer opened the pull request.

bernardd committed:
Bump version and update changelog. v2.2.9

avivek committed:
adding me to "aws" partition

LetThereBeDwight committed:
Add Athena Support in ca-central-1

dependabot[bot] committed:
Bump jason from 1.2.2 to 1.3.0

Bumps [jason](https://github.com/michalmuskala/jason) from 1.2.2 to 1.3.0.
- [Release notes](https://github.com/michalmuskala/jason/releases)
- [Changelog](https://github.com/michalmuskala/jason/blob/master/CHANGELOG.md)
- [Commits](https://github.com/michalmuskala/jason/compare/v1.2.2...v1.3.0)

---
updated-dependencies:
- dependency-name: jason
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Bump version and update changelog. v2.2.10

joeybaer committed:
feat(creds): support aws-sso configuration

kyleVsteger commented on lib/ex_aws/credentials_ini/file.ex:
```diff
@@ -1,17 +1,97 @@
+
+    defp rename_sso_credential_keys(role_credentials) do
+      # TODO: make this future proof and just convert all to snake case or be explicit?
+      Enum.into(role_credentials, %{}, fn {k, v} ->
+        case k do
+          "accessKeyId" -> {:access_key_id, v}
+          "expiration" -> {:expiration, v}
+          "secretAccessKey" -> {:secret_access_key, v}
+          "sessionToken" -> {:session_token, v}
+        end
+      end)
+    end
```
This will be a little less fragile in the event that more keys come back on role credentials
```suggestion
      Enum.reduce(role_credentials, %{}, fn 
          {"accessKeyId", v}, acc -> Map.put(acc, :access_key_id, v)
          {"expiration", v}, acc -> Map.put(acc, :expiration, v)
          {"secretAccessKey", v}, acc -> Map.put(acc, :secret_access_key, v)
          {"sessionToken", v}, acc -> Map.put(acc, :session_token, v)
          _, acc -> acc
        end
      end)
```

joeybaer replied:
💯  thanks @kyleVsteger

joeybaer committed:
feat(creds): improve durability of get-role-credentials

Co-authored-by: kyleVsteger <kylesteger@gmail.com>

bernardd committed:
Bump CI versions

bernardd committed:
roll back CI Erlang version to maintain support for elixir 1.7

bernardd committed:
Use latest elixir for dialyzer and format checks

lex-mala committed:
raise when bucket is nil

dependabot[bot] committed:
Bump ex_doc from 0.26.0 to 0.27.3

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.26.0 to 0.27.3.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.26.0...v0.27.3)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Merge pull request #854 from ex-aws/dependabot/hex/ex_doc-0.27.3

Bump ex_doc from 0.26.0 to 0.27.3

aymanosman committed:
Add eu-west-2 region to transcribe service

aymanosman committed:
Add all supported regions for transcribe

Taken from https://docs.aws.amazon.com/general/latest/gr/transcribe.html

bernardd committed:
Merge pull request #855 from aymanosman/main

Add all supported regions for transcribe

bernardd committed:
Merge pull request #853 from lex-mala/nil-bucket

Raise when no bucket is available

whatyouhide committed:
Add a few specs and polish docs here and there

dependabot[bot] committed:
Bump ex_doc from 0.27.3 to 0.28.0

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.27.3 to 0.28.0.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.27.3...v0.28.0)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Merge pull request #857 from ex-aws/dependabot/hex/ex_doc-0.28.0

Bump ex_doc from 0.27.3 to 0.28.0

bernardd committed:
Merge pull request #856 from whatyouhide/patch-1

Add a few specs and polish docs here and there

thecristen committed:
Add Location service to "aws" partition

Enables the usage of AWS Location Service for the relevant regions.
This includes the maps, places, geofencing, tracking, and routes
services, for nine supported regions.

bernardd committed:
Merge pull request #858 from thecristen/main

Add Location service to "aws" partition

Commit:
Support IMDSv2

dependabot[bot] committed:
Bump hackney from 1.18.0 to 1.18.1

Bumps [hackney](https://github.com/benoitc/hackney) from 1.18.0 to 1.18.1.
- [Release notes](https://github.com/benoitc/hackney/releases)
- [Changelog](https://github.com/benoitc/hackney/blob/master/NEWS.md)
- [Commits](https://github.com/benoitc/hackney/compare/1.18.0...1.18.1)

---
updated-dependencies:
- dependency-name: hackney
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Merge pull request #859 from ex-aws/dependabot/hex/hackney-1.18.1

Bump hackney from 1.18.0 to 1.18.1

Commit:
Fix tests

dependabot[bot] committed:
Bump ex_doc from 0.28.0 to 0.28.1

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.28.0 to 0.28.1.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.28.0...v0.28.1)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Merge pull request #861 from ex-aws/dependabot/hex/ex_doc-0.28.1

Bump ex_doc from 0.28.0 to 0.28.1

dependabot[bot] committed:
Bump ex_doc from 0.28.1 to 0.28.2

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.28.1 to 0.28.2.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/compare/v0.28.1...v0.28.2)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Merge pull request #862 from ex-aws/dependabot/hex/ex_doc-0.28.2

Bump ex_doc from 0.28.1 to 0.28.2

james-gibson-fr committed:
Debugging, syntax fix

james-gibson-fr committed:
Handle errors, pass correct security token

james-gibson-fr committed:
Fix expiration handling

james-gibson-fr committed:
add error atom to expiration error

james-gibson-fr committed:
Fix error handling in sso role cred request

james-gibson-fr committed:
Change error handling, some cleaning up

james-gibson-fr committed:
Mix format my changes

james-gibson-fr committed:
move compare into with

james-gibson-fr committed:
remove unused variables

joeybaer committed:
feat(awscli/sso): add sso endpoints, use request abstraction without signing

james-gibson-fr committed:
formatting, add minimal http config, add sso to govcloud

james-gibson-fr committed:
add http_opts to common config

james-gibson-fr committed:
address comments

james-gibson-fr committed:
mix format

james-gibson-fr committed:
stylistic change to rename/remove merge_credentials

bernardd committed:
Merge pull request #860 from justincjohnson/justincjohnson-imds-v2

Instance Metadata Service (IMDS) v2

james-gibson-fr committed:
Add, fix small tests

dependabot[bot] committed:
Bump ex_doc from 0.28.2 to 0.28.3

Bumps [ex_doc](https://github.com/elixir-lang/ex_doc) from 0.28.2 to 0.28.3.
- [Release notes](https://github.com/elixir-lang/ex_doc/releases)
- [Changelog](https://github.com/elixir-lang/ex_doc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/elixir-lang/ex_doc/commits)

---
updated-dependencies:
- dependency-name: ex_doc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

bernardd committed:
Merge pull request #864 from ex-aws/dependabot/hex/ex_doc-0.28.3

Bump ex_doc from 0.28.2 to 0.28.3

james-gibson-fr committed:
Merge remote-tracking branch 'ex_aws/main' into jg/aws-sso

joeybaer committed:
feat(sso): update gov endpoints
